casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -65,15 +107,32 @@ object ConflictSetMerger {
     val rejectionTargetF = (dc: Branch) => dc.map(cost).sum
     val optimalRejection = getOptimalRejection(rejectionOptions, rejectionTargetF)
     val toMerge          = branches diff optimalRejection
-    val rejected         = lateSet ++ rejectedAsDependents ++ optimalRejection.flatten
 
     for {
-      r                               <- Stopwatch.duration(toMerge.toList.flatten.traverse(stateChanges).map(_.combineAll))
+      baseMergeableChRes <- toMerge.flatten
+                             .map(mergeableChannels)
+                             .flatMap(_.keys)
+                             .toList
+                             .traverse(
+                               channelHash =>
+                                 convertToReadNumber(getData)
+                                   .apply(channelHash)
+                                   .map(res => (channelHash, res.getOrElse(0L)))
+                             )
+                             .map(_.toMap)
+      rejectedOverflowNegative = getMergedResultRejection(
+        toMerge,
+        baseMergeableChRes,
+        cost
+      )
+      ultimateMerge                   = branches.flatten diff optimalRejection.flatten diff rejectedOverflowNegative
```

Comment:
We need little more logic here. `optimalRejection` is from channel conflicts and must be combined with rejected by negative/overflow before remove from all branches.
The reason is that rejection in one group can influence in rejection in other. This means to again calculate similar combinations like in `computeRejectionOptions` for channel conflicts.

Reply:
Could you elaborate  little bit how to `combine optimalRejection  with rejected by negative/overflow`

Reply:
I am thinking maybe we don't need `optimalRejection ` here but `rejectionOptions` here.

Reply:
We might get better options in rejectOptions. But the combine logic is still something need to be polished.